src/linear_algebra/char_poly/coeff.lean

```diff
@@ -132,8 +132,14 @@ begin
   unfold polynomial.eval, unfold eval₂,
   transitivity polynomial.sum (mat_poly_equiv M) (λ (e : ℕ) (a : matrix n n R),
     (a * (scalar n) r ^ e) i j),
-  { unfold polynomial.sum, rw sum_apply, rw sum_apply, dsimp, refl, },
-  { simp_rw ← (scalar n).map_pow, simp_rw ← (matrix.scalar.commute _ _).eq,
+  { unfold polynomial.sum matrix,
+    -- `rw sum_apply` gets stuck here
+    convert sum_apply j _ _,
+    refine congr_fun _ j,
+    dsimp,
+    convert sum_apply i _ _, },
+  { simp_rw ←ring_hom.map_pow,
+    simp_rw ← (matrix.scalar.commute _ _).eq,
```

Comment:
It's a shame this proof became so awkward; I think `finset.sum_apply` can't see through `matrix` well enough to sort out typeclass unification. At any rate, this is the only casualty in this PR.

Reply:
I can't test it at the moment: does it help if instead of unfolding, you start the proof with `rw [polynomial.eval_eq_finset_sum, sum_apply, sum_apply]`?

Reply:
My conclusion was that I should just add a `matrix.sum_apply`, which has the benefit of not needing to see through `matrix` to use it.